homework/sophie_hw/metaphlan2_notes.md

```diff
@@ -0,0 +1,336 @@
+        * It worked!
+
+    `$ metaphlan2.py SRS014476-Supragingival_plaque.fasta.gz  --input_type fasta > SRS014476-Supragingival_plaque_profile.txt`
+
+    `$ less -S SRS014476-Supragingival_plaque_profile.txt`
+
+## Run on multiple cores
+*What does "multiple cores" mean?*
+
+`$ metaphlan2.py SRS014459-Stool.fasta.gz --input_type fasta --nproc 4 > SRS014459-Stool_profile.txt`
+
+* Similar problem to above, remove file (SRS014459-Stool.fasta.gz.bowtie2out.txt) and try again
+    * Error message below
+
+**Error message:**
+`OSError: Not a gzipped file (b'>H')`
+* Check .txt file, empty
+
+`$ less -S SRS014459-Stool_profile.txt`
+
+* Prof Google not helpful
+* Sent question to Lauren Tso & Kevin Bonham
+
+### Error trouble shooting
+1. *From Kevin* - Click download, rename to not have .gz, then zip
+
+`$ mv ~/Downloads/SRS*.fasta.gz ~/gitrepos/metaphlan_tutorial/`
+
+`$ mv SRS014476-Supragingival_plaque.fasta.gz SRS014476-Supragingival_plaque.fasta`
+
+`$ gzip SRS014476-Supragingival_plaque.fasta`
+
+`$ metaphlan2.py SRS014476-Supragingival_plaque.fasta.gz --input_type fasta --nproc 4 > SRS014459-Supragingival_plaque_profile.txt`
+
+2. *From Lauren* Delete all files, re-download everything using `$ curl -O`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014476-Supragingival_plaque.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014494-Posterior_fornix.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014459-Stool.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014464-Anterior_nares.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014470-Tongue_dorsum.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014472-Buccal_mucosa.fasta.gz`
+
+`$ mv ~/SRS*.fasta.gz ~/gitrepos/metaphlan_tutorial`
+*Only necessary if dir not ~/gitrepos/metaphlan_tutorial*
+
+## Run multiple Samples
+`$ metaphlan2.py SRS014464-Anterior_nares.fasta.gz --input_type fasta --nproc 4 > SRS014464-Anterior_nares_profile.txt`  
+
+`$ metaphlan2.py SRS014470-Tongue_dorsum.fasta.gz --input_type fasta --nproc 4 > SRS014470-Tongue_dorsum_profile.txt`
+
+`$ metaphlan2.py SRS014472-Buccal_mucosa.fasta.gz --input_type fasta --nproc 4 > SRS014472-Buccal_mucosa_profile.txt`
+
+`$ metaphlan2.py SRS014494-Posterior_fornix.fasta.gz --input_type fasta --nproc 4 > SRS014494-Posterior_fornix_profile.txt`
+
+* Create single tab-delimited table from output files
+`$ merge_metaphlan_tables.py *_profile.txt > merged_abundance_table.txt`
+
+`$ less -S merged_abundance_table.txt` :q
+
+# Vistualiza results
+
+## Create a heatmap with hclust2
+* hclust2 = plotting tool
```

Comment:
Well, technically it's the hierarchical clustering tool, but it also allows plotting.